ArduCopter/mode_autorotate.cpp

```diff
@@ -82,27 +81,68 @@ void ModeAutorotate::run()
         set_mode(copter.prev_control_mode, ModeReason::AUTOROTATION_BAILOUT);
     }
 
+    float pilot_roll, pilot_pitch, pilot_yaw_rate;
+    if (g2.arot._param_guided == 1){
+        pilot_roll = guided_roll;				
+    } else {
+        // Operator is in control of roll and yaw.  Controls act as if in stabilise flight mode.  Pitch 
+        // is controlled by speed-height controller.	
+        get_pilot_desired_lean_angles(pilot_roll, pilot_pitch, copter.aparm.angle_max, copter.aparm.angle_max);						
+    }
+		
+    float target_roll;
+
+    // Grab inertial velocity
+    const Vector3f& vel = inertial_nav.get_velocity_neu_cms();
+
+    // rotate roll, pitch input from north facing to vehicle's perspective
+    float roll_vel =  vel.y * ahrs.cos_yaw() - vel.x * ahrs.sin_yaw(); // body roll vel
+    float pitch_vel = vel.y * ahrs.sin_yaw() + vel.x * ahrs.cos_yaw(); // body pitch vel
+
+    // gain scheduling for yaw
+    float pitch_vel2 = MIN(fabsf(pitch_vel), 2000);
+    pilot_yaw_rate = ((float)pilot_roll/1.0f) * (1.0f - (pitch_vel2 / 5000.0f)) * g2.acro_y_rate; //****THIS NEEDS FIXED*****
+
+    roll_vel = constrain_float(roll_vel, -560.0f, 560.0f);
```

Comment:
We shouldn't hard code constrains.  We also don't need to.  The user defined input shaping will be done in the attitude control library